viewing_party/party.py

```diff
@@ -1,23 +1,210 @@
-# ------------- WAVE 1 --------------------
+# # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    new_movie = create_movie(movie["title"],movie["genre"],movie["rating"])
+    user_data["watched"] = [new_movie]
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    new_movie = create_movie(movie["title"],movie["genre"],movie["rating"])
+    user_data["watchlist"] = [new_movie]
+
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:      
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum_of_ratings = 0.0
+    ratings_avg = 0.0
+    length = len(user_data["watched"])
+    if length > 0:
+        for i in range(length):
+            sum_of_ratings += user_data["watched"][i]["rating"]      
+            
+        ratings_avg = sum_of_ratings/length
 
+        return ratings_avg
+    else:
+        return 0.0
+    
+def get_most_watched_genre(user_data):
+    count_dict = {}
+    cntr = 0
+    next_genre = ""
+    length = len(user_data["watched"])
+
+    if length > 0:
+        for i in range(length):
+            next_genre += user_data["watched"][i]["genre"] 
+        
+            if next_genre in count_dict:
+                cntr += 1
+            else:
+                cntr = 1
+            
+            count_dict.update({next_genre:cntr})    

```

Comment:
when using methods like update/max write a short comment on what it is doing in your code.